File: back/src/main/java/com/project/booktime/model/entity/Library.java
package com.project.booktime.model.entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document("library")
public class Library {

    @Id
    private String id;
    private String userId;
    private List<String> bookIdList;

    public Library(String userId, List<String> bookIdList) {
        this.userId = userId;
        this.bookIdList = bookIdList;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public List<String> getBookIdList() {
        return bookIdList;
    }

    public void setBookIdList(List<String> bookIdList) {
        this.bookIdList = bookIdList;
    }
}

File: back-node/src/models/Library.ts
import { ObjectId } from "mongoose";
import List from "../utils/List";

export default class User 
{
    private _id: ObjectId;
    private userId: ObjectId;
    private bookIdList: List<ObjectId>;
 
    constructor(userId: ObjectId, bookIdList: List<ObjectId>)
    {
        this.userId = userId;
        this.bookIdList = bookIdList;
    }

    public getId(): ObjectId {
        return this._id;
    }

    public setId(value: ObjectId) {
        this._id = value;
    }

    public getUserId(): ObjectId {
        return this.userId;
    }

    public setUserId(value: ObjectId) {
        this.userId = value;
    }

    public getBookIdList(): List<ObjectId> {
        return this.bookIdList;
    }

    public setBookIdList(value: List<ObjectId>) {
        this.bookIdList = value;
    }

}